packages/subgraph/src/utils.ts

```diff
@@ -80,6 +80,15 @@ export function initializeEventEntity(
     entity.set("timestamp", Value.fromBigInt(event.block.timestamp));
     entity.set("transactionHash", Value.fromBytes(event.transaction.hash));
     entity.set("gasPrice", Value.fromBigInt(event.transaction.gasPrice));
+    if (event.receipt) {
+        // @note if we don't cast event.receipt as ethereum.TransactionReceipt,
+        // we get a compile error because it thinks event.receipt is:
+        // ethereum.TransactionReceipt | null
+        const receipt = event.receipt as ethereum.TransactionReceipt;
+        entity.set("gasUsed", Value.fromBigInt(receipt.gasUsed));
+    } else {
+        entity.set("gasUsed", Value.fromBigInt(BIG_INT_ZERO));
```

Comment:
If the receipt should always be present (i.e. specified in the `.yaml`) then I'd prefer throwing in general.

Reply:
it will throw because when we go and try to save the event, this field is non-nullable